### DISCUSSION TITLE
GitHub Alternitive Suggestion

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#2071] [Ideas]
I would like to suggest considering switching the Vaultwarden project to a self-hosted instance of GitLab, or even just on the main server. The reason I suggest this is that GitHub is part of the Big Tech mafia, owned by Microsoft.

Given a large motivating factor of installing Vaultwarden over using Bitwarden, or even LastPass, Dashlane, etc... is the inherently secure and private aspect of self-hosting. The sentiment and risk involved in using a Big Tech owned platform is unsettling to me, and I'm sure many others agree.

What is the position of the Vaultwarden development community on this matter? Is this something you are concerned about? 

Thank you.

**Comment by jjlin** [comment]
Self-hosting is not inherently secure -- it just shifts the security responsibility to the self-hoster. Do you have specific and tangible concerns aside from not liking that GitHub is owned by a member of the "Big Tech mafia"?

**Reply by Masgalor** [reply]
While I totally agree, that selfhosting a git service to gain an increase in security dosent make sense, I may add concerns some organizations / projects see in GitHub.

Microsoft often blindly follows DMCA or other copyright related take-down requests available in US law and suspends or deletes repos without mediation or fact checking those claims.
The same way Microsoft does region-block users from countries that are under an US trade embargo for example.

I dont think those are reasons to ditch GitHub, but for some it is enough to at least mirror their project to other providers e.g. Codeberg in the EU.

**Comment by ctschach** [comment]
There are so many big projects hosted on GitHub - also a lot of security relevant projects. Is anything serious happens, we will know it...much faster than we would hear about a self hosted server get's compromised.

I know, it's owned by M$, but still under the supervision by so many people. The only evil thing that they could do is shutting down the servers...which I don't expect to ever happen.

I would even go that far that I trust GitHub more than a privately hosted project. Just because it's more visible. If someone would add some evil code to Vaultwarden I expect that it will be found much faster here then if it's hidden somewhere.

**Comment by redpixy** [comment]
I'm sorry but i don't see how Microsoft's ownership of Github is going to affect the source code of vaultwarden? As for the risk, maybe you can elaborate further. 

No offence but it seems you have some vendetta with members of the "Big Tech mafia".

**Comment by BlackDex** [comment]
Well, all our code on Github is visible to everyone, same as the build-process of the images, hell even most commits are GPG-Signed.
And saying that Github is owned by "Big Tech Mafia" does that also mean you are not using Linux then? Since Microsoft is a member of the Linux Foundation?

You are always free to fork this repo or mirror it to Gitlab, or any server you want if that is what you want.
But don't forget that all of the Bitwarden code, like web vault, mobile client etc... is also on Github.

**Reply by RealOrangeOne** [reply]
As are Rust, Docker, and about 90% of the other dependencies of the project.
